dev/integration_tests/web/web/index_with_blocked_service_workers.html

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE HTML>
+<!-- Copyright 2014 The Flutter Authors. All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. -->
+<html>
+<head>
+  <meta charset="UTF-8">
+  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
+
+  <title>Web Test</title>
+  <!-- iOS meta tags & icons -->
+  <meta name="apple-mobile-web-app-capable" content="yes">
+  <meta name="apple-mobile-web-app-status-bar-style" content="black">
+  <meta name="apple-mobile-web-app-title" content="Web Test">
+  <link rel="manifest" href="manifest.json">
+
+  <script>
+    // This is to break the serviceWorker registration, and make it throw a DOMException!
+    // Test for issue https://github.com/flutter/flutter/issues/103972
+    window.navigator.serviceWorker.register = (url) => {
+      console.error('Failed to register/update a ServiceWorker for scope ', url,
+          ': Storage access is restricted in this context due to user settings or private browsing mode.');
+      return Promise.reject(new DOMException('The operation is insecure.'));
+    }
```

Comment:
Haha I love JavaScript. PATCH THE PLATFORM!

Reply:
I wonder what will happen first: the FF issue getting fixed, or Chromium stopping me from writing on the `register` function of the `ServiceWorkerContainer` object :P